### PR TITLE
Adding book lending example

### PR DESCRIPTION
**Pull request by tsurdilo**
Signed-off-by: Tihomir Surdilovic <tsurdilo@redhat.com>

**Many thanks for submitting your Pull Request :heart:!**

**Please specify parts this PR updates:**

- [ ] Specification
- [ ] Schema
- [ x] Examples
- [ ] Extensions
- [ ] Roadmap
- [ ] Use Cases
- [ ] Community
- [ ] TCK
- [ ] Other

**What this PR does / why we need it**:
Adds a new example - Book Lending
**Special notes for reviewers**:

**Additional information (if needed):**

### CODE REVIEW

**Comment by tsurdilo** [comment]
Parsed example: https://github.com/tsurdilo/specification/tree/booklendingexample/examples#book-lending

File: examples/README.md [hunk 1]
```diff
@@ -3485,4 +3486,267 @@ functions:
...
+
+### Book Lending
+
+#### Description
+
+In this example we want to create a book lending workflow. The workflow starts when a lender
+submits a book request (via event "Book Lending Request Event").
+The workflow describes our business logic around lending a book, from checking its current availability,
+to waiting on the lenders response if the book is currently not available, to checking out the book and notifying
+the lender.
```
Comment (ricardozanini) [inline examples/README.md:3499]:
```suggestion
In this example, we want to create a book lending workflow. The workflow starts when a lender
submits a book lending request (via event "Book Lending Request Event").
The workflow describes our business logic around lending a book, from checking its current availability,
to waiting on the lender's response if the book is currently not available, to checking out the book and notifying
the lender.
```

File: examples/README.md [hunk 2]
```diff
@@ -3485,4 +3486,267 @@ functions:
...
+        "address": " ... "
+        "phone: " ... "
+    }
+}
+```
+
+where the "book" property defines the book to be lended out, and the "lender" property provides info 
+about the person wanting to lend the book.
+
+For the sake of the example we assume the functions and event definitions are defined in separate json files.
```
Comment (ricardozanini) [inline examples/README.md:3520]:
```suggestion
where the "book" property defines the book to be lent out, and the "lender" property provides info 
about the person wanting to lend the book.

For the sake of the example, we assume the functions and event definitions are defined in separate JSON files.
```

**Review by ricardozanini — APPROVED** [review]
Are we still using JSON Path as expressions?

**Comment by tsurdilo** [comment]
@ricardozanini 
**Are we still using JSON Path as expressions?** : these are jq expressions (simple selections)

**Review by ricardozanini — APPROVED** [review]
Dammit, I forgot to fix this on my past review. Sorry.

**Comment by ricardozanini** [comment]
> @ricardozanini
> **Are we still using JSON Path as expressions?** : these are jq expressions (simple selections)

Brain fart.

File: examples/README.md [hunk 3]
```diff
@@ -3485,4 +3486,267 @@ functions:
...
+            }
+         ]
+      },
+      {
+         "name": "Request Hold",
+         "type": "operation",
+         "actions": [
+            {
+               "functionRef": {
+                  "refName": "Request fold for lender",
```
Comment (jorgenj) [inline examples/README.md:3624]: nit; 'Request hold for lender'

**Comment by tsurdilo** [comment]
rebased and fixes per comments

**Pull request merged**